src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```diff
@@ -0,0 +1,252 @@
+                is not set.
+
+        Returns:
+            The orchestrator run id.
+        """
+        try:
+            return os.environ[ENV_ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID]
+        except KeyError:
+            raise RuntimeError(
+                "Unable to read run id from environment variable "
+                f"{ENV_ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID}."
+            )
+
+    @property
+    @abstractmethod
+    def cloud(self) -> sky.clouds.Cloud:
+        """The type of sky cloud to use.
+
+        Returns:
+            A `sky.clouds.Cloud` instance.
+        """
+
+    def setup_credentials(self) -> Any:
+        """Set up credentials for the orchestrator.
+
+        Returns:
+            A client authentication object, dependant on the cloud.
+        """
+        connector = self.get_connector()
+        return connector.connect()
+
+    def get_setup(self, stack: Optional["Stack"]) -> Optional[str]:
+        """Run to set up the sky job.
+
+        Returns:
+            A `setup` string.
+        """
+        return None
+
+    def prepare_or_run_pipeline(
+        self,
+        deployment: "PipelineDeploymentResponseModel",
+        stack: "Stack",
+        environment: Dict[str, str],
+    ) -> Any:
+        """Runs all pipeline steps in Skypilot containers.
+
+        Args:
+            deployment: The pipeline deployment to prepare or run.
+            stack: The stack the pipeline will run on.
+            environment: Environment variables to set in the orchestration
+                environment.
+
+        Raises:
+            RuntimeError: If a step fails.
+        """
+        if deployment.schedule:
+            logger.warning(
+                "Skypilot Orchestrator currently does not support the"
+                "use of schedules. The `schedule` will be ignored "
+                "and the pipeline will be run immediately."
+            )
+
+        # Set up some variables for configuration
+        orchestrator_run_id = str(uuid4())
+        environment[
+            ENV_ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID
+        ] = orchestrator_run_id
+
+        settings = cast(
+            SkypilotBaseOrchestratorSettings,
+            self.get_settings(deployment),
+        )
+
+        entrypoint = PipelineEntrypointConfiguration.get_entrypoint_command()
+        entrypoint_str = " ".join(entrypoint)
+        arguments = PipelineEntrypointConfiguration.get_entrypoint_arguments(
+            deployment_id=deployment.id
+        )
+        arguments_str = " ".join(arguments)
+
+        # Set up docker run command
+        image = self.get_image(deployment=deployment)
+        docker_environment_str = " ".join(
+            f"-e {k}={v}" for k, v in environment.items()
+        )
+
+        start_time = time.time()
+
+        instance_type = settings.instance_type or self.DEFAULT_INSTANCE_TYPE
+
+        # Set up credentials
+        self.setup_credentials()
```

Comment:
@stefannica Need your help here. On my server (hamza.zenml.io) I have connectors connected but they are not being picked up (I tried on the GCP case). Maybe we need to sit down and see how we can side-inject into skypilot

Reply:
After a discussion with @stefannica and an exploration of the code, the decision is made to implement basic versions of the [_configure_local_client](https://github.com/zenml-io/zenml/blob/main/src/zenml/service_connectors/service_connector.py#L292) method for all cloud providers

Reply:
Is this already implemented? Is it the option to write the local config files with connectors that were added in this PR?